benchmarks/WolfSheep/wolf_sheep.py

```diff
@@ -49,55 +149,69 @@ def __init__(
             grass_regrowth_time: How long it takes for a grass patch to regrow
                                  once it is eaten
             sheep_gain_from_food: Energy sheep gain from grass, if enabled.
+            moore:
         """
         super().__init__(seed=seed)
         # Set parameters
         self.height = height
         self.width = width
         self.initial_sheep = initial_sheep
         self.initial_wolves = initial_wolves
-        self.sheep_reproduce = sheep_reproduce
-        self.wolf_reproduce = wolf_reproduce
-        self.wolf_gain_from_food = wolf_gain_from_food
         self.grass_regrowth_time = grass_regrowth_time
-        self.sheep_gain_from_food = sheep_gain_from_food
 
         self.schedule = RandomActivationByType(self)
-        self.grid = MultiGrid(self.height, self.width, torus=False)
+        self.grid = OrthogonalGrid(
+            self.height, self.width, moore=moore, torus=False, capacity=math.inf
+        )
 
         # Create sheep:
-        for _i in range(self.initial_sheep):
+        for _ in range(self.initial_sheep):
             pos = (
                 self.random.randrange(self.width),
                 self.random.randrange(self.height),
             )
-            energy = self.random.randrange(2 * self.sheep_gain_from_food)
-            sheep = Sheep(self.next_id(), pos, self, True, energy)
-            self.grid.place_agent(sheep, pos)
+            energy = self.random.randrange(2 * sheep_gain_from_food)
+            sheep = Sheep(
+                self.next_id(), self, energy, sheep_reproduce, sheep_gain_from_food
+            )
+            sheep.move_to(self.grid.cells[pos])
             self.schedule.add(sheep)
 
         # Create wolves
-        for _i in range(self.initial_wolves):
+        for _ in range(self.initial_wolves):
             pos = (
                 self.random.randrange(self.width),
                 self.random.randrange(self.height),
             )
-            energy = self.random.randrange(2 * self.wolf_gain_from_food)
-            wolf = Wolf(self.next_id(), pos, self, True, energy)
-            self.grid.place_agent(wolf, pos)
+            energy = self.random.randrange(2 * wolf_gain_from_food)
+            wolf = Wolf(
+                self.next_id(), self, energy, wolf_reproduce, wolf_gain_from_food
+            )
+            wolf.move_to(self.grid.cells[pos])
```

Comment:
Thank you for agreeing to https://github.com/projectmesa/mesa/issues/1953#issuecomment-1891098411 & to be as intuitive as NetLogo.

Reply:
Once we added a CellAgent, adding a dedicated move method was a small step. Random walks are now just a one-liner as well.